### PR TITLE
feat(components/HeaderBar): app switcher in brand section

### PR DESCRIPTION
**Pull request by ybaskaran**
**What is the problem this PR is trying to solve?**
As per the recent changes in guidelines, the app switcher should be displayed on the application name with a dropdown. Ref: http://guidelines.talend.com/document/92132#/navigation-layout/app-header/app-switcher

**What is the chosen solution to this problem?**
App switcher removed from products section and relocated to the brand section in the header bar

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[x] This PR introduces a breaking change**

Only for QA. Tests on App switcher will fail, they should use the new AppHeaderBar component-object, included in this PR.

### CODE REVIEW

**Comment by ybaskaran** [comment]
I see some styling issues with dropdown as per guidelines such as the width of the dropdown and the icon sizes especially when the app name is long, I have already requested the changes from Marielle and waiting for her response.
as_per_guidelines
modified

**Comment by ybaskaran** [comment]
brand icons and alignments are changed to adhere with the style guidelines

**Comment by frassinier** [comment]
It's a breaking change! Please fill the related file.

File: packages/theme/src/theme/_visual-helpers.scss [hunk 1]
```diff
@@ -38,16 +38,16 @@
...
 			display: inline-flex;
 			align-items: center;
 
 			&:before {
 				display: block;
 				content: '';
 				margin-right: $padding-small;
-				height: $svg-md-size;
-				width: $svg-md-size;
+				height: 2rem;
```
Comment (frassinier) [inline packages/theme/src/theme/_visual-helpers.scss]: Both could be replaced by `$navbar-brand-logo-height` I guess

File: packages/components/stories/HeaderBar.js [hunk 2]
```diff
@@ -98,7 +98,24 @@ const props = {
 	brand: {
 		id: 'header-brand',
 		label: 'Example App Name',
-		onClick: action('onApplicationNameClick'),
+		items: [
```
Comment (frassinier) [inline packages/components/stories/HeaderBar.js]: Breaking change spotted

File: packages/components/src/HeaderBar/HeaderBar.scss [hunk 3]
```diff
@@ -79,6 +79,17 @@ $background-color-on-hover: shade($brand-primary, 30);
 		width: $svg-lg-size;
 	}
 
+	.tc-header-bar-brand + :global(.dropdown-menu) :global(.tc-svg-icon) {
```
Comment (frassinier) [inline packages/components/src/HeaderBar/HeaderBar.scss]: or `:global(.dropdown-menu .tc-svg-icon)`

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 4]
```diff
@@ -40,19 +40,29 @@ function Brand({ label, isSeparated, getComponent, ...props }) {
...
+		props && props.items ? (
+			<Renderers.ActionDropdown
+				bsStyle="link"
+				className={theme['tc-header-bar-brand']}
+				tooltipPlacement="bottom"
+				label={label}
+				{...props}
+			/>
+		) : (
+			<Renderers.Action
```
Comment (frassinier) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: Don't repeat yourself

```javascript
const Component = props && props.items ? Renderers.ActionDropdown : Renderers.Action;
const renderOption = (<Component 
    bsStyle="link"
	className={theme['tc-header-bar-brand']}
	tooltipPlacement="bottom"
	label={label}
	{...props} 
/>);
```

File: packages/components/stories/HeaderBar.js (hunk 2 above)
Comment (jsomsanith) [reply packages/components/stories/HeaderBar.js]: Added breaking change log in the description (to report in wiki on merge)

**Pull request merged**